src/main/java/io/cryostat/agent/CryostatClient.java

```diff
@@ -354,7 +360,10 @@ private CountingInputStream getRecordingInputStream(Path filePath) throws IOExce
     }
 
     private String selfMatchExpression(URI callback) {
-        return String.format("target.connectUrl == \"%s\"", callback);
+        return String.format(
+                "target.connectUrl == \"%s\" && target.jvmId == \"%s\" &&"
+                        + " target.annotations.platform[\"INSTANCE_ID\"] == \"%s\"",
```

Comment:
Any reason to use platform but not `cryostat` annotations?

Reply:
https://github.com/cryostatio/cryostat/blob/bee2ffbc21c21fb231608e6c02e401fa0e951313/src/main/java/io/cryostat/platform/ServiceRef.java#L164

https://github.com/cryostatio/cryostat/blob/bee2ffbc21c21fb231608e6c02e401fa0e951313/src/main/java/io/cryostat/platform/ServiceRef.java#L210

The `cryostat` annotations are meant to have only this restricted key set, none of which really match what this is doing. This agent instance ID is somewhat similar conceptually to the hash that k8s uses to make ex. Pods unique, and those would also go in the `platform` annotations, so it makes sense to put it here from both angles IMO.